### PR TITLE
Setup cloudinary #26

### PR DESCRIPTION
**Pull request by uche-okoroafor**
### What this PR does (required):
Setup Cloudinary   #26 
- Allow for uploading of images 

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

https://user-images.githubusercontent.com/71725892/139826769-aaa646c2-8a70-4b23-9806-230d3197081d.mp4

### Any information needed to test this feature (required):
- In your postman or Insomnia, you will need to create a POST request to http://localhost:3001/image/upload
- In the request body you should select form-data and add "imageName" as the key name and select an image file.    
- If the image is uploaded successfully, you can check the response from the postman.

### CODE REVIEW

File: server/controllers/image.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                data: { uploadStatus, imageDetails },
+              });
+            })
+            .catch((error) => {
+              res.status(500).json({
+                success: false,
+                message: `Error creating image in the database: ${error.message}`,
+              });
+            });
+        });
```
Comment (ezeilo-su) [inline server/controllers/image.js:51]: Please consider using `async/await` to be more readable.

**Review by ezeilo-su — COMMENTED** [review]
Good work with this one :+1:  I've left a few nice-to-have comments for you to take a look at.

File: server/routes/image.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const { uploadImage } = require("../controllers/image");
+const { validateUserIdParams } = require("../middleware/validateRouteParams");
+const router = require("express").Router();
+
+router.post(
+  "/upload",
+  validateUserIdParams,
+  upload.imageUpload.any(),
+  uploadImage
+);
```
Comment (ezeilo-su) [inline server/routes/image.js:11]: This should be a protected route. Use the `protect` middleware.

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Please take a look at those comments :+1:

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please address merge conflicts!

**Review by moffatethan — APPROVED** [review]

**Comment by moffatethan** [comment]
@uche-okoroafor Please merge this in!

**Pull request merged**